### PR TITLE
[IMP] hr_recruitment: Replace Blacklist mechanism with Job platforms

### PR DESCRIPTION
**Pull request by jugj-odoo**
task-3167538

### CODE REVIEW

**Comment by sofiagvaladze** [comment]
Don't forget to link with the task :bulb: and if it does not exist yet, to create it

**Review by sofiagvaladze — CHANGES_REQUESTED** [review]
@jugj-odoo thanks for your work :raised_hands: 

some stylistic and one upgrade comment

also the commit message, it's empty :monocle_face:

File: addons/hr_recruitment/models/hr_job_platform.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+from odoo import fields, models
+
+
+class JobPlatform(models.Model):
+    _name = "hr.job.platform"
+    _description = 'Job Platforms'
+
+    name = fields.Char(required=True)
+    email = fields.Char(required=True, help="Applications received from this Email won't be linked to a contact."
+                                            "There will be no email address set on the Applicant either.")
```
[the remark this answers is not in the document]
Comment (jugj-odoo) [reply addons/hr_recruitment/models/hr_job_platform.py:12]: there's automatic line breaks

**Comment by sofiagvaladze** [comment]
robodoo r+

File: addons/hr_recruitment/models/hr_applicant.py [hunk 2]
```diff
@@ -634,10 +627,14 @@ def message_new(self, msg, custom_values=None):
...
             'partner_name': partner_name or email_from_normalized,
         }
-        if msg.get('from') and not self._email_is_blacklisted(msg.get('from')):
+        job_platform = self.env['hr.job.platform'].search([('email', '=', email_from_normalized)], limit=1)
+        if msg.get('from') and not job_platform:
             defaults['email_from'] = msg.get('from')
             defaults['partner_id'] = msg.get('author_id', False)
-        if msg.get('email_from') and self._email_is_blacklisted(msg.get('email_from')):
+        if msg.get('email_from') and job_platform:
+            subject_pattern = re.compile(job_platform.regex)
```
Comment (KangOl) [inline addons/hr_recruitment/models/hr_applicant.py:635]: Will it work if the regex is not filled?

Comment (jugj-odoo) [reply addons/hr_recruitment/models/hr_applicant.py:635]: it won't match anything yeah

Comment (jugj-odoo) [reply addons/hr_recruitment/models/hr_applicant.py]: updated it to use an empty string if not filled, it won't match anything

**Comment by sofiagvaladze** [comment]
robodoo r+

**Pull request closed without merging**